client/allocrunner/taskrunner/getter/sandbox_test.go

```diff
@@ -51,3 +51,27 @@ func TestSandbox_Get_http(t *testing.T) {
 	must.NoError(t, err)
 	must.StrContains(t, string(b), "module github.com/hashicorp/go-set")
 }
+
+func TestSandbox_Get_insecure_http(t *testing.T) {
+	testutil.RequireRoot(t)
+	logger := testlog.HCLogger(t)
+
+	ac := artifactConfig(10 * time.Second)
+	sbox := New(ac, logger)
+
+	_, taskDir := SetupDir(t)
+	env := noopTaskEnv(taskDir)
+
+	artifact := &structs.TaskArtifact{
+		GetterSource: "https://expired.badssl.com",
```

Comment:
As neat as this service is, we probably shouldn't make requests to 3rdparty sites in out tests / CI runs. Instead I think we can get away with just starting an http server using [httptest.NewTLSServer](https://pkg.go.dev/net/http/httptest#NewTLSServer) and using our own `http.Client` which won't have the generated certs, and should fail as expected.